concrete/src/Cookie/CookieJar.php

```diff
@@ -1,113 +1,180 @@
 <?php
+
 namespace Concrete\Core\Cookie;
 
 use Concrete\Core\Http\Request;
-use Symfony\Component\HttpFoundation\Cookie as CookieObject;
 
+/**
+ * A class that holds operations performed on both request and response cookies.
+ *
+ * To work only on request cookies, use the Request class.
+ * To work only on response cookies, use the ResponseCookieJar class.
+ */
 class CookieJar
 {
-    protected $cookies = array();
-    protected $clearedCookies = array();
+    /**
+     * The object containing the request cookies.
+     *
+     * @var \Concrete\Core\Http\Request
+     */
     protected $request;
 
     /**
-     * Adds a CookieObject to the cookie pantry.
+     * The object containing the response cookies.
      *
-     * @param string $name The cookie name
-     * @param string|null $value The value of the cookie
-     * @param int $expire The number of seconds until the cookie expires
-     * @param string $path The path for the cookie
-     * @param null|string $domain The domain the cookie is available to
-     * @param bool $secure whether the cookie should only be transmitted over a HTTPS connection from the client
-     * @param bool $httpOnly Whether the cookie will be made accessible only through the HTTP protocol
+     * @var \Concrete\Core\Cookie\ResponseCookieJar
+     */
+    protected $responseCookieJar;
+
+    /**
+     * Initialize the instance.
      *
-     * @return \Symfony\Component\HttpFoundation\Cookie
+     * @param Request $request the object containing the request cookies
+     * @param ResponseCookieJar $responseCookieJar the object containing the response cookies
      */
-    public function set(
-        $name,
-        $value = null,
-        $expire = 0,
-        $path = '/',
-        $domain = null,
-        $secure = false,
-        $httpOnly = true
-    ) {
-        $cookie = new CookieObject($name, $value, $expire, $path, $domain, $secure, $httpOnly);
-        $this->add($cookie);
-
-        return $cookie;
+    public function __construct(Request $request, ResponseCookieJar $responseCookieJar)
+    {
+        $this->setRequest($request);
+        $this->responseCookieJar = $responseCookieJar;
     }
 
     /**
-     * Adds a CookieObject to the array of cookies for the object.
+     * Get the object containing the response cookies.
      *
-     * @param CookieObject $cookie
+     * @return \Concrete\Core\Cookie\ResponseCookieJar
      */
-    public function add($cookie)
+    public function getResponseCookiesJar()
```

Comment:
This should either be `getResponseCookies` or `getResponseCookieJar`. In my opinion we shouldn't denote the type in the method handle, so I'd vote for `->getResponseCookies()`

Reply:
Yeah, I agree with @KorvinSzanto on this, I'd definitely vote for getResponseCookies() that returns the responsecookiejar object.

Reply:
> Yeah, I agree with @KorvinSzanto on this, I'd definitely vote for getResponseCookies() that returns the responsecookiejar object.

[PR updated](https://github.com/concrete5/concrete5/pull/6894/commits/2b6c88e08f449bdedc31457903d4a7a116decaff).